scipy/interpolate/_bsplines.py

```diff
@@ -1765,6 +1765,9 @@ def make_lsq_spline(x, y, t, k=3, w=None, axis=0, check_finite=True, *, method="
     axis = normalize_axis_index(axis, y.ndim)
 
     y = np.moveaxis(y, axis, 0)    # now internally interp axis is zero
+    if not y.flags.c_contiguous:
+        # C routines in _dierckx currently require C contiguity
+        y = y.copy()
```

Comment:
I see it's the default, but might be best to make it explicit since it's _not_ the default of `np.copy`.
```suggestion
        y = y.copy(order='C')
```

Reply:
Agreed, done